### PR TITLE
Redirects user to lecture page and grabs lecture key from URL

### PR DESCRIPTION
**Pull request by jaysaleh**
This pull request redirects each lecture to the `lecture-view.html` and grabs the key from the URL.

### CODE REVIEW

File: src/main/webapp/lectures.js [hunk 1]
```diff
@@ -1,3 +1,7 @@
+/* Used to redirect users to the lecture view page, while storing lecture key in
+ * the URL. */
+const LINK_PREFIX = '/lecture-view.html?key=';
```
Comment (n-wach) [inline src/main/webapp/lectures.js:3]: I'd recommend using `URL.searchParams` to construct this.  For instance:
```
  const url = new URL('/lecture-view.html', window.location.origin);
  url.searchParams.append('key', key);
```

File: src/main/webapp/lectures.js [hunk 2]
```diff
@@ -23,8 +27,7 @@ function createLectureListItem(lecture) {
   const listItem = document.createElement('li');
   const lectureLink = document.createElement('a');
 
-  // TODO: Redirect to specific lecture page.
-  lectureLink.href = lecture.videoUrl;
+  lectureLink.href = LINK_PREFIX + lecture.key.id;
```
Comment (n-wach) [inline src/main/webapp/lectures.js:30]: Use `URL` object, see above

File: src/main/webapp/view.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+/* Number of characters until lecture ID in URL. */
+const URL_ID_SPACING = 5;
```
Comment (n-wach) [inline src/main/webapp/view.js:2]: Where does the 5 come from? `?key=`?

Comment (n-wach) [reply src/main/webapp/view.js:2]: I think we should just use URL and URLSearchParams to do this parsing/generating

Comment (jaysaleh) [reply src/main/webapp/view.js:2]: i agree, ill make the change

File: src/main/webapp/lectures.js (hunk 1 above)
Comment (jaysaleh) [reply src/main/webapp/lectures.js:3]: fixed

File: src/main/webapp/lectures.js (hunk 2 above)
Comment (jaysaleh) [reply src/main/webapp/lectures.js:30]: fixed

**Pull request merged**